src/SprykerEco/Zed/Unzer/Business/Checkout/UnzerPostSaveCheckoutHookExecutor.php

```diff
@@ -42,27 +43,43 @@ public function __construct(
      * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
      * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponseTransfer
      *
-     * @return void
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
      */
-    public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer): void
+    public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer): CheckoutResponseTransfer
     {
-        if ($quoteTransfer->getPayment() === null) {
-            return;
-        }
-
-        if ($quoteTransfer->getPaymentOrFail()->getPaymentProvider() !== SharedUnzerConfig::PAYMENT_PROVIDER_NAME) {
-            return;
+        if (!$this->hasUnzerPayment($quoteTransfer)) {
+            return $checkoutResponseTransfer;
         }
 
         $paymentMethod = $quoteTransfer->getPaymentOrFail()->getPaymentSelectionOrFail();
         $paymentProcessor = $this->unzerPaymentProcessorResolver->resolvePaymentProcessor($paymentMethod);
         $unzerPaymentTransfer = $paymentProcessor->processOrderPayment($quoteTransfer, $checkoutResponseTransfer->getSaveOrderOrFail());
 
-        $checkoutResponseTransfer
+        if ($unzerPaymentTransfer->getErrors()->count() !== 0) {
+            foreach ($unzerPaymentTransfer->getErrors() as $unzerPaymentErrorTransfer) {
```

Comment:
please move to the separate method and return it's result here

Reply:
Fixed.